src/common/transformations/src/transformations/convert_precision.cpp

```diff
@@ -1065,10 +1065,15 @@ bool extend_reverse_type(const std::shared_ptr<ov::Node>& node, const precisions
 
 template <typename src_type, typename dst_type>
 inline dst_type convert_value(src_type val) {
-    if (val > std::numeric_limits<dst_type>::max()) {
-        return std::numeric_limits<dst_type>::max();
-    } else if (val < std::numeric_limits<dst_type>::lowest()) {
-        return std::numeric_limits<dst_type>::lowest();
+    if (std::numeric_limits<src_type>::max() > std::numeric_limits<dst_type>::max()) {
```

Comment:
```suggestion
    if constexpr (std::numeric_limits<src_type>::max() > std::numeric_limits<dst_type>::max()) {
```

Reply:
Tried, it doesn't work because of bfloat16 our custom implementation